packages/react/src/text-input/TextInput.tsx

```diff
@@ -113,18 +111,18 @@ export const textInputStyles = ({
 			width: '100%',
 		}),
 
-		...(invalid && {
-			backgroundColor: boxPalette.systemErrorMuted,
-			borderColor: boxPalette.systemError,
-		}),
-
 		...(multiline && {
 			paddingTop: mapSpacing(0.5),
 			paddingBottom: mapSpacing(0.5),
 			height: 'auto',
 			minHeight: '6rem',
 		}),
 
+		'&[aria-invalid="true"]': {
```

Comment:
Since `GroupedFields` needs to set `aria-invalid` on elements, I've changed all form inputs to be styled off the presence of this attribute. Being less reliant on JS (and using CSS) is much better imo… as well as being the only way I could get Grouped Fields to work anyway :)